src/StaticAnalyser.php

```diff
@@ -391,4 +390,30 @@ private function parseUseStatement(&$tokens, &$token, $parseContext)
 
         return $statements;
     }
+
+    private function extractTypeAndNextToken(array &$tokens, Context $parseContext): array
+    {
+        $type = UNDEFINED;
+        $nullable = false;
+        $token = $this->nextToken($tokens, $parseContext);
+
+        if ($token[0] === T_STATIC) {
+            $token = $this->nextToken($tokens, $parseContext);
+        }
+
+        if ($token === '?') { // nullable type
+            $nullable = true;
+            $token = $this->nextToken($tokens, $parseContext);
+        }
+
+        // drill down namespace segments to basename property type declaration
+        while ($token[0] === T_NS_SEPARATOR || $token[0] === T_STRING) {
```

Comment:
This should read 
```
while (in_array($token[0], [T_NS_SEPARATOR, T_STRING, T_ARRAY])) {
            if (in_array($token[0], [T_STRING, T_ARRAY])) {```

Reply:
although now I've thought about it, there's very little to be gained from the `array` type. The openApi document is going to need an 'items' property to be considered valid, and most codesniffer rules require you to have a var annotation for array types to specify what the items will be. I think it might be better if we just fix the handling of @var annotations for array types

Reply:
If we just add `T_ARRAY` to the `while` this will fix the problem reported by @SlvrEagle23 
If we add `T_ARRAY` to the if statement, it will override the `@var` annotation as you've designed it to do.
Actually why does native trump `@var`?

Reply:
Fixed in https://github.com/jpuck/swagger-php/pull/3

Reply:
fabulous, thank you! sorry I was away from Github for a few days. so happy to see this open source carrying on without me 😄